nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -1010,8 +1010,15 @@ public DocumentRef getParentDocumentRef(DocumentRef docRef) {
 
     @Override
     public DocumentModel getParentDocument(DocumentRef docRef) {
-        Document doc = resolveReference(docRef);
-        Document parentDoc = doc.getParent();
+        Document parentDoc;
+        if (docRef != null && docRef.type() == DocumentRef.PATH) {
+            String targetPath = ((String) docRef.reference());
+            String parentPath = targetPath.substring(0, targetPath.lastIndexOf("/"));
+            parentDoc = resolveReference( new PathRef(parentPath));
+        } else {
+            Document doc = resolveReference(docRef);
+            parentDoc = doc.getParent();
+        }
```

Comment:
Finally, I wonder if this change should also be applied to:
```
#getParentDocumentRef
#getParentDocuments
#getParentDocumentRefs
```
In which case you could refactor part of the code.
@efge what do you think?

Reply:
Agreed, we should be consistent.

Reply:
Actually the method 
```#getParentDocuments```
Seems to be not consistent with the others since it includes the child document in the returned list (if I 'm not wrong)

Reply:
Yes if there's no clear use case for now go ahead and skip it.